add subtitle to perf issue feed/details

DominikB2014 opened the pull request.

DominikB2014 committed:
add subtitle to perf issue feed

udameli commented on static/app/utils/events.tsx:
```diff
@@ -152,7 +152,7 @@ export function getTitle(
       // back to the computed title for these.
       return {
-        title: customTitle ?? (metadata.message || event.title),
+        title: customTitle ?? (metadata.message || title),
```
i think for perf issues the title will be coming from `metadata.title`

DominikB2014 replied:
Hmm, Originally it was returned from event.title, was the not intended?
<img width="248" alt="image" src="https://user-images.githubusercontent.com/44422760/189405339-b80f1f94-a186-4a84-ade3-55d125f34e76.png">

DominikB2014 replied:
The title in this line is actually unrelated to perf issues, its just because I destructured the object on line 106.
The title form perf issue should come from `customTitle` which is set by `metadata.title`

DominikB2014 replied:
Grabbing from metadata now!

DominikB2014 committed:
fix tsc

evanpurkhiser commented on static/app/types/event.tsx:
```diff
@@ -395,6 +395,7 @@ interface EventBase {
   device?: Record<string, any>;
   endTimestamp?: number;
   groupID?: string;
+  issueCategory?: 'performance' | 'error';
```
Should we use the `IssueCategory` type here?

DominikB2014 replied:
Yup! tbh didn't know that was a type. Thanks!

evanpurkhiser commented on static/app/utils/events.tsx:
```diff
@@ -162,9 +162,15 @@ export function getTitle(
+    case EventOrGroupType.TRANSACTION:
+      return {
+        title: customTitle ?? title,
+        subtitle: event.issueCategory === 'performance' ? culprit : '',
```
```suggestion
        subtitle: event.issueCategory === IssueCategory.PERFORMANCE ? culprit : '',
```

DominikB2014 committed:
add types and grab title from metadata

DominikB2014 committed:
Merge branch 'master' into fix-perf-issue-feed-subtitle

danecando committed:
fix(replays): Improve performance of DOM Events tab open (#38456)

Defers running the mutations code until after the tab change to reduce lag between clicking the tab and seeing any changes. DOM Events tab will open and show a spinner until the mutations have loaded.

Improvement for #37549

Before:
https://user-images.githubusercontent.com/3721977/188647887-c8d8d45b-0ca2-4eb3-a8dc-4013dd6c44fb.mp4

After: 
https://user-images.githubusercontent.com/3721977/188647915-5d65d668-6a68-420e-83e9-0b7a2b68262f.mp4

evanpurkhiser committed:
fix(js): Pass `group_id` when fetching specific events (#38649)

DominikB2014 committed:
added unit tests to ensure transaction name exists in document

DominikB2014 commented on static/app/components/eventOrGroupTitle.spec.tsx:
```diff
@@ -0,0 +1,115 @@
+import {render, screen} from 'sentry-test/reactTestingLibrary';
```
Just did a typescript conversion

DominikB2014 committed:
fix

malwilley committed:
fix(issue-stream): Disable stack trace tooltip for performance issues [UP-146] (#38601)

Added a check for `issueCategory` before rendering `<StackTraceDetails
/>`

However, this introduced some styling issues that I had to resolve. The
stack trace applied `fontSizeLarge`, so skipping that component caused
performance issues to have a smaller title. I chose to move that font
size to the top-level div and override the font-size from any consumers
that need a different font-size. These were:

- `<IssueLink />` (used in the activity feed tooltips)
- `<EventDetailsContent />` (used in the discover event page header)
- `<GroupHeader />` (used in the issue details page)

DominikB2014 committed:
Revert "fix(issue-stream): Disable stack trace tooltip for performance issues [UP-146] (#38601)"

This reverts commit d3bce9759bf6d9fb58662660ce8d61e2eb4b4898.

DominikB2014 committed:
Revert "fix(js): Pass `group_id` when fetching specific events (#38649)"

This reverts commit 42483dd62674dfe245676f8b3446f85281504a3a.

DominikB2014 committed:
Revert "fix(replays): Improve performance of DOM Events tab open (#38456)"

This reverts commit e90e391b6b13e206471bf55e179e074c3ec452c5.

DominikB2014 committed:
Merge remote-tracking branch 'origin/master' into fix-perf-issue-feed-subtitle

DominikB2014 committed:
resolve merge conflicts

DominikB2014 commented:
Had some weirdness with resolving merge conflicts, its all good now lol
